### PR TITLE
Featute: Adding method to validate ingredient availability

### PR DESCRIPTION
**Pull request by eapg**
* Adding method to validate ingredient availability
* Adding test for validate ingredient availability

This method will take two arguments(the **ingredient** to validate and the **quantity to use**) , with the **ingredient.id** the method filter inside the repository the **inventory_ingredient** that match with the **ingredient id**, after that it compare its **quantity** with the **quantity to use**, if **quantity** is higher that **quantity to use** the method return **True** otherwise return **False**.

### CODE REVIEW

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py [hunk 1]
```diff
@@ -45,3 +45,22 @@ def get_by_ingredient_id(self, ingredient):
             inventory_ingredients,
         )
         return list(inventory_ingredient_by_ingredient_id)
+
+    def validate_ingredient_availability(
+        self, inventory_id, ingredient, quantity_to_use
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/inventory_ingredient_repository_impl.py:50]:
```suggestion
        self, inventory_id, ingredient_id, quantity_to_use
```

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py [hunk 2]
```diff
@@ -45,3 +45,22 @@ def get_by_ingredient_id(self, ingredient):
...
+    ):
+
+        inventory_ingredients = self.get_all()
+        ingredient_to_validate = list(
+            filter(
+                (
+                    lambda inventory_ingredient: inventory_ingredient.ingredient.id
+                    == ingredient.id
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/inventory_ingredient_repository_impl.py:58]:
```suggestion
                    == ingredient_id
```

File: src/lib/repositories/inventory_ingredient_repository.py [hunk 3]
```diff
@@ -8,3 +8,8 @@ class InventoryIngredientRepository(GenericRepository, metaclass=ABCMeta):
     @abstractmethod
     def get_by_ingredient_id(self, ingredient):
         pass
+
+    @abstractmethod
+    def validate_ingredient_availability(self, inventory_id, ingredient, quantity_to_use):
```
Comment (reynoldmorel) [inline src/lib/repositories/inventory_ingredient_repository.py:13]:
```suggestion
    def validate_ingredient_availability(self, inventory_id, ingredient_id, quantity_to_use):
```

File: src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py [hunk 4]
```diff
@@ -134,3 +135,23 @@ def test_get_by_ingredient_id_successfully(self):
...
+        inventory_1 = build_inventory(inventory_id=1)
+        ingredient_1 = build_ingredient(ingredient_id=1, name="ingredient test")
+
+        inventory_ingredient_1 = build_inventory_ingredient(
+            ingredient=ingredient_1, inventory=inventory_1, ingredient_quantity=10
+        )
+        inventory_ingredient_repository.add(inventory_ingredient_1)
+        self.assertTrue(
+            inventory_ingredient_repository.validate_ingredient_availability(
+                inventory_1.id, ingredient_1, 5
```
Comment (reynoldmorel) [inline src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py:150]:
```suggestion
                inventory_1.id, ingredient_1.id, 5
```

File: src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py [hunk 5]
```diff
@@ -134,3 +135,23 @@ def test_get_by_ingredient_id_successfully(self):
...
+            )
+        )
+        self.assertFalse(
+            inventory_ingredient_repository.validate_ingredient_availability(
+                inventory_1.id, ingredient_1, 15
```
Comment (reynoldmorel) [inline src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py:155]:
```suggestion
                inventory_1.id, ingredient_1.id, 15
```

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

**Comment by eapg** [comment]
Done.

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py (hunk 1 above)
Comment (eapg) [reply src/lib/repositories/impl/inventory_ingredient_repository_impl.py]: done

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py (hunk 2 above)
Comment (eapg) [reply src/lib/repositories/impl/inventory_ingredient_repository_impl.py]: done

File: src/lib/repositories/inventory_ingredient_repository.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/inventory_ingredient_repository.py]: done

File: src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py (hunk 4 above)
Comment (eapg) [reply src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py]: done

File: src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py (hunk 5 above)
Comment (eapg) [reply src/tests/lib/repositories/impl/inventory_ingredient_repository_impl_test.py]: done

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**